ci/jenkins/utils.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    echo "====== Checking installed Golang version ======"
+    antrea_golang_version="go$(grep -E "^go\ " ${WORKSPACE}/go.mod | awk '{print $2}' | cut -d. -f1-2)"
+
+    curl -s "https://go.dev/dl/?mode=json&include=all" | jq -c -r '.[]' | while read i; do
+        version=$(echo ${i} | jq .version | tr -d '"')
+        if [[ "${version}" =~ "${antrea_golang_version}" ]]; then
+            if [ -d "/usr/local/${antrea_golang_version}" ]; then
+                current_version=$(/usr/local/${antrea_golang_version}/bin/go version | awk '{print $3}')
+                if [[ "${version}" == "${current_version}" ]]; then
+                    echo "====== Golang ${version} is installed on the testbed, no need to download ======"
+                    switch_golang "${version}"
+                    break
+                fi
+            fi
+            echo "====== Installing golang version "${version}" ======"
+            install_golang "${version}"
+            switch_golang "${version}"
+            break
+        fi
+    done
+}
+
+function install_golang() {
+    golang_version=$1
+    echo "====== Downloading Golang ${golang_version} ======"
+    curl https://dl.google.com/go/${golang_version}.linux-amd64.tar.gz -o /tmp/${golang_version}.linux-amd64.tar.gz
+    sudo rm -rf /tmp/go || true
+    sudo tar xf /tmp/${golang_version}.linux-amd64.tar.gz -C /tmp/
+    golang_version=$(echo ${golang_version} | cut -d. -f1-2)
+    sudo rm -rf /usr/local/${golang_version} || true
```

Comment:
A bit concern on jenkins user modifying things under /usr/local. Usually only root or sudoers can do this. I'm not willing to give the jenkins user write permission on paths outside of home dir. This is because a PR potentially can have or import malicious code. It's better putting everything in jenkins home, or run things in container such as `make docker-bin`, `make docker-test-unit`.

Maybe it's possible to install Golang to jenkins ~/go, and configure jenkins user's environment variable to point GOPATH, PATH to paths under ~/go.